Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,329 @@
+            _whereClause = new WhereClause();
+        }
+
+        public WhereClause VisitWhere(LambdaExpression whereClause)
+        {
+            _whereClause.ExpNode = Extract(whereClause.Body);
+            var json = JsonConvert.SerializeObject(_whereClause, formatting: Formatting.Indented);
+            Visit(whereClause.Body);
+            return _whereClause;
+        }
+
+        private ExpressionNode Extract(Expression node)
+        {
+            var realmLinqExpression = Visit(node) as RealmLinqExpression;
+            return realmLinqExpression.ExpressionNode;
+        }
+
+        protected override Expression VisitBinary(BinaryExpression be)
+        {
+            ExpressionNode returnNode;
+            if (be.NodeType == ExpressionType.AndAlso)
+            {
+                var andNode = new AndNode();
+                andNode.Left = Extract(be.Left);
+                andNode.Right = Extract(be.Right);
+                returnNode = andNode;
+            }
+            else if (be.NodeType == ExpressionType.OrElse)
+            {
+                var orNode = new OrNode();
+                orNode.Left = Extract(be.Left);
+                orNode.Right = Extract(be.Right);
+                returnNode = orNode;
+            }
+            else
+            {
+                ComparisonNode comparisonNode;
+                switch (be.NodeType)
+                {
+                    case ExpressionType.Equal:
+                        comparisonNode = new EqualityNode();
+                        break;
+                    case ExpressionType.NotEqual:
+                        comparisonNode = new NotEqualNode();
+                        break;
+                    case ExpressionType.LessThan:
+                        comparisonNode = new LtNode();
+                        break;
+                    case ExpressionType.LessThanOrEqual:
+                        comparisonNode = new LteNode();
+                        break;
+                    case ExpressionType.GreaterThan:
+                        comparisonNode = new GtNode();
+                        break;
+                    case ExpressionType.GreaterThanOrEqual:
+                        comparisonNode = new GteNode();
+                        break;
+                    default:
+                        throw new NotSupportedException($"The binary operator '{be.NodeType}' is not supported");
+                }
+
+                if (be.Left is MemberExpression me)
+                {
+                    if (me.Expression != null && me.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        var leftName = GetColumnName(me, me.NodeType);
+
+                        comparisonNode.Left.Value = leftName;
+                        comparisonNode.Left.Kind = "property";
+                        // TODO: Possible with switch statment in our current .NET version?
+                        if (me.Type == typeof(float))
```

Comment:
This code is repeated in the same way some lines down. This is generally a sign that it could be extracted to another method.